Examples

pacostas opened the pull request:
For clients
* mysql (main structure)
* mysql2 (doesnt follow the main structure, but it will once is decided/reviewed)
* odbc-mysql (doesnt follow the main structure, but it will once is decided/reviewed)
* pg (main structure)
* odbc-psql (doesnt follow the main structure, but it will once is decided/reviewed)
* rhea [only instructions no sample app]

mhdawson commented on README.md:
```diff
@@ -1,3 +1,171 @@
 # kube-service-bindings
 
-A repository with examples utilizing [kube-service-binding](https://www.npmjs.com/package/kube-service-bindings) npm package:
+A repository with examples utilizing [kube-service-bindings](https://www.npmjs.com/package/kube-service-bindings) npm package:
```
```suggestion
A repository with examples and links to examples utilizing [kube-service-bindings](https://www.npmjs.com/package/kube-service-bindings) npm package:
```

mhdawson commented on README.md:
```diff
@@ -1,3 +1,171 @@
+
+- KAFKA
+  - node-rdkafka
+  - kafkajs
```
```suggestion
  - [node-rdkafka](https://github.com/nodeshift-starters/reactive-example/tree/node-rdkafka)
  - [kafkajs](https://github.com/nodeshift-starters/reactive-example)
```

mhdawson commented on README.md:
```diff
@@ -1,3 +1,171 @@
+  - [mysql](./src/mysql/README.md)
+
+# Basic instructions
+
```
The examples listed need some common configuration along with some additional components needed by the
exmple itself. In terms of common configuration they all need an OpenShift cluster and the service binding operator installed. As an example of additional components, the example showing how to use MYSQL with
kube-service-bindings needs a MYSQL database installed.

This section explains how to install those common and additional components. The README.md for each of the examples within this repo (not the external examples) which reference these instructions and tell you specifically what you need to install and in what order.

pacostas replied:
Yes, exactly.
Definitely I would like to rename the title of this part from "Basic Instructions" to "Common Configuration" if you agree.

mhdawson commented on README.md:
````diff
@@ -1,3 +1,171 @@
+   oc apply -f deploy/cr.yaml
+   ```
+
+## Deploy PostgreSQL - Crunchy DB in Openshift
````
Some of the Graphics in this section don't seem to line up correctly for me.

mhdawson commented on README.md:
```diff
@@ -1,3 +1,171 @@
+
+[install on all namespaces](./readme-assets/binding2.png)
+
+## Install Crunchy DB Operator
```
Some of the Graphics in this section don't seem to line up correctly for me.

mhdawson commented on README.md:
```diff
@@ -1,3 +1,171 @@
+  - [mysql](./src/mysql/README.md)
+
+# Basic instructions
+
```
It would also be good to expand my suggestion that you need to install and then also deploy an instance. I think it would be good to keep those two steps together in the same section, possibly as sub-sections.

mhdawson replied:
More comments as I read further down.
I think in this section we should explain the general flow

1) isntall the general stuff
2) install the specific stuff needed for an exmple
3) deploy and run the example

If 2) is not listed in the example itself then having a table that shows which example needs what would be good.

mhdawson commented on README.md:
```diff
@@ -1,3 +1,171 @@
+
+## Connecting Node.js app using service binding operator
+
+Simply by dragging a line between the deployed app and DB cluster you should be able to share credentials between those two, as shown on below image
```
```suggestion
Simply by dragging a line between the deployed app and the additional component used by the example (for example DB cluster) you should be able to share credentials between those two. The image below shows doing this with a database:
```

mhdawson commented on src/mongodb/README.md:
```diff
@@ -0,0 +1,51 @@
+1.  On the form, select first MongoDB Atlas Cloud Database on the Database provider drop down menu and then fill the rest fields -> Import.
+1.  In case of help for filling out the Account Credentials, follow this link, which provides detailed instructions for each field. https://access.redhat.com/documentation/en-us/red_hat_openshift_database_access/1/html-single/quick_start_guide/index#find-your-mongodb-atlas-account-credentials_rhoda-qsg
+
+## Deploying mongoDB database as Cloud-Hosted Database
```
Not sure if in this section or the earlier one, but we need to enable the IP filter in Atlast as well right?

mhdawson commented on src/rhea/README.md:
```diff
@@ -0,0 +1,54 @@
+
+## Deploy Node.js
+
+- with Nodeshift [Instructions](../../README.md#deploy-nodejs-application-with-nodeshift)
```
I think you might have missed checking in the files for this example? There is only a README.md

mhdawson replied:
I see you mentioned that above.

mhdawson commented:
I like the format where you have common instructions and refer to them with links.  The last part where you run the sample might need a bit more specific info/instructions through.

I think the bindings directories are not needed, right? They add a lot of files.  If they are useful for tests (although I've not seen that)

mhdawson commented on src/mongodb/server.js:
```diff
@@ -0,0 +1,208 @@
+    url
+  };
+}
+
```
I think since these examples are focussed on kube-service-bindings we could simplify them by just removing the non-service bindings path.

mhdawson commented on src/mysql2/README.md:
```diff
@@ -0,0 +1,3 @@
+# MySQL - mysql2 client
+
+The steps are exactly the same as on mongodb [example](../mysql/README.md)
```
```suggestion
The steps are exactly the same as on mysql[example](../mysql/README.md)
```

mhdawson commented on src/pg/server.js:
```diff
@@ -0,0 +1,146 @@
+  } else if (req.url === "/ready" && req.method === "GET") {
+    res.writeHead(200).end();
+  } else if (req.url === "/live" && req.method === "GET") {
+    res.writeHead(200).end();
```
If possible would be nice if the common code was the same across the examples. in some places this is two lines and in others like this one, just one.

mhdawson commented:
Will also need a licence file and possibly some other boilerplate files.

pacostas commented:
> I like the format where you have common instructions and refer to them with links. The last part where you run the sample might need a bit more specific info/instructions through.
> 
> I think the bindings directories are not needed, right? They add a lot of files. If they are useful for tests (although I've not seen that)

Yes indeed binding folders are not necessary, I deleted them.
The part of running the app I was about to forget. I'll add it once we decide on the UI/CRUD operations/In general the functionality of the app.

pacostas commented:
> Will also need a licence file and possibly some other boilerplate files.

Added MIT license, if you think its good

mhdawson commented on README.md:
```diff
@@ -1,3 +1,189 @@
+  - [mysql](./src/mysql/README.md)
+
+# Basic instructions
+
```
```suggestion
The examples listed need some common configuration along with some additional components needed by the
example itself. In terms of common configuration they all need an OpenShift cluster and the service binding operator installed. As an example of additional components, the example showing how to use MYSQL with
kube-service-bindings needs a MYSQL database installed.

This section explains how to install those common and additional components. The README.md for each of the examples within this repo (not the external examples) which reference these instructions and tell you specifically what you need to install and in what order.
```

pacostas committed:
Review 1


updating readme for mysql2 client

adding common node modules on root level

adding bindings files as exposed from percona operator

adding sample app using mysql2 client

adding sample app using odbc driver to connect to mysql

Initial readme file

adding vscode conf for mysql2 client

refactoring code to use updated service binding

mysql configuring eslint

adding odbc server

adding nodemon

finishing odbc example worker

renaming odbc example to odbc-mysql

adding odbc-psql example draft

adding credentials proper on odbc-pg example

mysql2 example additions

mysql2 example adding docker file

adding mysql client example

fixup! adding odbc-psql example draft

odbc mysql updating kube service binding package

adding ubi8 image for odbc examples

docs: adding more info on readme file

removing express dependency

removing express from mysql example

removing body-parser express from general dependencies

using yum package manager

docker image removing installations

using fedora image with registry url

fixing connection string

using npm i --production on installation

excluding package-lock otherwise new dependecies of kube-service-binding are not installed

fixup! fixing connection string

correcting dockerfile

adding bindings from crunchy psql operator

docker file using npm i instead of npm ci --productino

testing - changing connection string

adding pg example

adding isntructions on how to deploy a psql with crunchy db

updating eslint

configuring ssl

chore: reducing npm packages

fix: removing unmapped option

fix: updating connection configs for odbc mysql

docs: updating readme instructions

fix: updating dockerfile to use ubi8

feat: adding docker file and baseImage.yaml for psql

feat: adding baseImage.yaml for mysql

feat: adding odbc-base image

feat: adding setup docs on psql odbc

fix: updating bindings for psql

fix: updating readme with more instructions on how to deploy in openshift

feat: adding yaml files for deployment with oc command

refactor: credentials fallback on odbc psql example

refactor: using global eslint and prettier configuration

feat: adding mongodb example

fix: including properly mongoClient

feat: adding mongoose example

fix: mongo credentials passing

fix: catch block

fix: using the official mongodb example

feat: adding routes for crud operations in mongodb

fix: adding CRUD on mongoose example

fix: curating md files

fix: updating docs and dependencies

fix: adding insructions for mysql client

fix: crud operation on mysql client

fix: updating readme files of mysql, odbc-psql and pg

feat: curating the pg example

feat: adding post get and get all route

feat: using fruits exampel (draft commit)

fix: finishing pg example

fix: renaming on mongo app.js to server.js

fix: finishing mongodb example

fix: updating mongodb readme instructions

fix: updating readme of mongodb, mysql, mysql2, odbc-mysql, odbc-psql

fix: odbc-psql base image url change

fix: updating rhea instructions

Update README.md

Co-authored-by: Michael Dawson <mdawson@devrus.com>
Update README.md

Co-authored-by: Michael Dawson <mdawson@devrus.com>
fix: update readme.md graphics

Update README.md

Co-authored-by: Michael Dawson <mdawson@devrus.com>
fix: removing unecessasry binding folders

fix: cropping images

Create LICENSE.md
refactor: adding routes, handlers and controllers

mongodb fallback url

Update README.md

Co-authored-by: Michael Dawson <mdawson@devrus.com>

pacostas committed:
Review 2

Co-authored-by: Michael Dawson <mdawson@devrus.com>
refactor: finishing pg example

Update src/mysql2/README.md

fix: odbc-psql example

fix: finished mysq example

fix: finished mysql2 example

fix: finished odbc-mysql example

fix: finishing mongoose example

fix: updating readme with further instructions

fix: updating port to be on 8080

fix: pg example final fixes

fix: pgsql odbc example

fix: mysql example final testing

fix: psql removing namespace from yaml files

fix: adding build config progress instructions. Using cr-minimal for percona

fix: odbc-mysql adding yaml files for base images and build configs

refactor: renaming yaml files of build configs on psql

fix: odbc-mysql queries using MySQL syntax

fix: mysql-odbc example finish

pacostas committed:
forgot to hit save

mhdawson commented on README.md:
```diff
@@ -1,3 +1,248 @@
 # kube-service-bindings
 
-A repository with examples utilizing [kube-service-binding](https://www.npmjs.com/package/kube-service-bindings) npm package:
+A repository with examples and links to examples utilizing [kube-service-bindings](https://www.npmjs.com/package/kube-service-bindings) npm package:
```
```suggestion
A repository with examples and links to examples utilizing the [kube-service-bindings](https://www.npmjs.com/package/kube-service-bindings) npm package:
```

mhdawson committed:
Update README.md

mhdawson reviewed (approved):
I have a number of larger comments, but I'll open issues in this repo for those and we can handles those one by one.
